File: EcmApiClient/src/com/pfizer/ecmapi/client/models/SearchQuery.java
package com.pfizer.ecmapi.client.models;

import com.thoughtworks.xstream.annotations.XStreamAlias;

@XStreamAlias("SearchQuery")
public class SearchQuery {
	@XStreamAlias("SearchType")
	private String searchType;
	@XStreamAlias("FolderId")
	private String folderId;
	@XStreamAlias("Query")
	private String query;
	@XStreamAlias("MaxNumberFiles")
	private int maxNumberFiles = -1;
	@XStreamAlias("StartingMetadataTableNumber")
	private int startingMetadataTableNumber = -1;
	@XStreamAlias("NumberOfMetadataTablesToSearch")
	private int numberOfMetadataTablesToSearch = -1;
	@XStreamAlias("StartDate")
	private String startDate;
	@XStreamAlias("EndDate")
	private String endDate;
	@XStreamAlias("IncludeFileDetails")
	private boolean includeFileDetails = true;
	
	
	public SearchQuery() {}

	public String getSearchType() {
		return searchType;
	}

	public void setSearchType(String searchType) {
		this.searchType = searchType;
	}

	public String getFolderId() {
		return folderId;
	}

	public void setFolderId(String folderId) {
		this.folderId = folderId;
	}

	public String getQuery() {
		return query;
	}

	public void setQuery(String query) {
		this.query = query;
	}

	public int getMaxNumberFiles() {
		return maxNumberFiles;
	}

	public void setMaxNumberFiles(int maxNumberFiles) {
		this.maxNumberFiles = maxNumberFiles;
	}

	public int getStartingMetadataTableNumber() {
		return startingMetadataTableNumber;
	}

	public void setStartingMetadataTableNumber(int startingMetadataTableNumber) {
		this.startingMetadataTableNumber = startingMetadataTableNumber;
	}

	public int getNumberOfMetadataTablesToSearch() {
		return numberOfMetadataTablesToSearch;
	}

	public void setNumberOfMetadataTablesToSearch(int numberOfMetadataTablesToSearch) {
		this.numberOfMetadataTablesToSearch = numberOfMetadataTablesToSearch;
	}

	public String getStartDate() {
		return startDate;
	}

	public void setStartDate(String startDate) {
		this.startDate = startDate;
	}

	public String getEndDate() {
		return endDate;
	}

	public void setEndDate(String endDate) {
		this.endDate = endDate;
	}
	
	public boolean isIncludeFileDetails() {
		return includeFileDetails;
	}

	public void setIncludeFileDetails(boolean includeFileDetails) {
		this.includeFileDetails = includeFileDetails;
	}

}

File: EcmApiRestService/RestService/Models/SearchQuery.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Runtime.Serialization;
using System.Web;

namespace EcmapiRestService.Models
{
    [DataContract(Namespace = "")]
    public class SearchQuery
    {
        [DataMember(Order = 1)]
        public string SearchType { get; set; }
        [DataMember(Order = 2)]
        public string FolderId { get; set; }
        [DataMember(Order = 3)]
        public string Query { get; set; }
        [DataMember(Order = 4)]
        public int MaxNumberFiles { get; set; }
        [DataMember(Order = 5)]
        public int StartingMetadataTableNumber { get; set; }
        [DataMember(Order = 6)]
        public int NumberOfMetadataTablesToSearch { get; set; }
        [DataMember(Order = 7)]
        public string StartDate { get; set; }       // For example: 2015-02-09 10:43:54
        [DataMember(Order = 8)]
        public string EndDate { get; set; }         // For example: 2015-02-09 17:00:00
        [DataMember(Order = 9)]
        public bool IncludeFileDetails { get; set; }
    }

}